data-preprocessing/video-download-upload/video-download-upload.py

```diff
@@ -0,0 +1,137 @@
+#This python script reads video links from a given sheet in an excel document, 
+#downloads the videos and saves the videos to a google drive folder
+#K.T Yeboah | 27-02-2023| 11:44am
+
+
+#import relevant libaries
+import pandas as pd
+from pytube import YouTube
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaFileUpload
+from pathlib import Path
+import json
+import argparse
+import os
+
+
+
+#Brief usage function
+def usage():
+    print("Usage: python nsmq-video-download-upload.py [options]")
+    print("")
+    print("Options:")
+    print("  --h                  Print this usage message")
+
+#Function to read instruction file containing necessary parameters/values for this script to run
+def readInstructionFile():
+    with open('ins.json', 'r') as ins:
+        data = json.load(ins)
+    #print(data)
+    return data
+
+#Function to read excel sheet and return stream/data
+def readExcelSheet(doc, sheet):
+    df = pd.read_excel(doc, sheet)
+    return df
+
+#Function to download video
+def downloadVideo(link, localVidDir):
+    try:
+        yt = YouTube(link)
+        video = yt.streams.filter(adaptive=True, file_extension='mp4').order_by('resolution').desc().first()
+        video.download(localVidDir)
+        print(f"video '{link}' downloaded succesfully")
+    except Exception as e:
+        print(f'Error downloading {link}: {e}')
+
+#Function to validate json instruction file
+def check_json_file(filepath):
+    required_keys = ["excel_doc_path", "sheet_name", "video_folder_path", "video_header_name",
+                     "google_drive_folder_name", "google_drive_folder_ID", "service_account_path"]
+    
+    with open(filepath) as f:
+        data = json.load(f)
+    
+    for key in required_keys:
+        if key not in data:
+            raise ValueError("bad JSON instruction file")
+    
+    return True
+
+
+def __main__(args):
+#Validate Json file and throw error if file is bad
+    try:
+        check_json_file("ins.json")
```

Comment:
this means the `ins.json` file must be in the same folder as the `video-download-upload.py` file. May be worth it mention that in the README.md as well

Reply:
Mentioned that